tools/rustfmt/srcs/lib.rs

```diff
@@ -5,16 +5,20 @@ use std::path::{Path, PathBuf};
 /// The expected extension of rustfmt manifest files generated by `rustfmt_aspect`.
 pub const RUSTFMT_MANIFEST_EXTENSION: &str = "rustfmt";
 
-/// Generate an absolute path to a file without resolving symlinks
-fn absolutify_existing<T: AsRef<Path>>(path: &T) -> std::io::Result<PathBuf> {
-    let absolute_path = if path.as_ref().is_absolute() {
-        path.as_ref().to_owned()
-    } else {
-        std::env::current_dir()
-            .expect("Failed to get working directory")
-            .join(path)
-    };
-    std::fs::metadata(&absolute_path).map(|_| absolute_path)
+pub fn current_dir_name() -> PathBuf {
+    PathBuf::from(std::env::current_dir().unwrap().file_name().unwrap())
+}
+
+pub fn from_slash(p: PathBuf) -> PathBuf {
+    let s = p
+        .to_string_lossy()
+        .chars()
+        .map(|c| match c {
+            '/' => std::path::MAIN_SEPARATOR,
+            c => c,
+        })
+        .collect::<String>();
+    PathBuf::from(s)
```

Comment:
Essentially https://docs.rs/path-slash/0.1.4/src/path_slash/lib.rs.html#302; I would have also not liked to re-implement a directory walking function. Can this repo have third-party dependencies on these crates?

Reply:
There are ways to use 3rd party crates but for this particular tool I'd say it's not desirable since it'd require users to load the dependencies before the target can be used. Maybe there's an easy path forward there that I'm not thinking about but my current thought is that this target should remain free of 3rd party deps 😅